src/client/components/DataManager/GlobalProvider.js

```diff
@@ -0,0 +1,51 @@
+import React, { createContext } from "react";
+import useFetch from "./useFetch";
+
+export const GlobalContext = createContext();
+
+export const GlobalProvider = ({ children }) => {
+  const { data: meals } = useFetch("/api/meals");
+  const { data: mealsWithBookings } = useFetch("/api/mealsPlus");
+  const { data: reviews } = useFetch("/api/reviews");
```

Comment:
Yeah, this is what I was worried was going on :D 
This feels like typical pre-mature optimization without really considering the data you have.

What if a new meal is added in the API? The client won't get this unless they hard reload their browser tab (since it's a SPA, normal page navigation won't cause a refetch). Or if a new review is added. 
Even if a booking is made this data is never updated unless a hard refresh happens.

I'll speculate that you're maybe worried about pulling this data form the API for each page that needs it, since it feels like it doesn't change that often?

If that's the case, there are better ways to solve this problem and it's called caching. You could add a in memory cache in your `useFetch` handler, that only calls your API every 5 minutes for example.
There are many challenges with client side caching, the more common way is simply to cache on the server side. With a CDN in front, the cost of the network request isn't so bad and the cache saves your server from processing and from requesting data from the database (which is usually the slow part).

Similarly, if you're worried about having to repeatedly do processing on this data in the client you can consider two options: 
1) Change your API to match your most common use case
2) Use something like reselect to memoize computation between renders (if it's really needed).